crates/wasm-mutate/src/mutators/peephole/eggsy/encoder.rs

```diff
@@ -89,80 +56,61 @@ impl Encoder {
             Some(entry.return_type.clone())
         };
 
-        // Return the type of a sibling
-        //      op
-        //    /    \
-        // t1.x      t?.y
-        // => t? = t1
-        // the sibling is calculated by increasing the index in the parent
-        // overflow are prevented by calculating the remainder on the number of children
-        // this will prevent: index overflow errors, return the type of the same operand
-        let anysibling = |parentid: Id, index_at_parent: Option<usize>| {
-            let rootlang = &nodes[usize::from(parentid)];
-            match rootlang {
-                Lang::Add(operands)
-                | Lang::Eq(operands)
-                | Lang::Sub(operands)
-                | Lang::GtS(operands)
-                | Lang::Mul(operands)
-                | Lang::GtU(operands)
-                | Lang::Xor(operands)
-                | Lang::LtS(operands)
-                | Lang::And(operands)
-                | Lang::LtU(operands)
-                | Lang::Or(operands)
-                | Lang::GeS(operands)
-                | Lang::Shl(operands)
-                | Lang::GeU(operands)
-                | Lang::ShrS(operands)
-                | Lang::LeS(operands)
-                | Lang::ShrU(operands)
-                | Lang::LeU(operands) => {
-                    let siblingidx = (usize::from(index_at_parent?) + 1) % operands.len();
-                    let sibling = &operands[siblingidx];
-                    gettpe(*sibling)
-                }
-                _ => None,
+        let types = nodes
+            .iter()
+            .map(|_| None)
+            .collect::<Vec<Option<PrimitiveTypeInfo>>>();
+        let types = RefCell::new(types);
+        let changed = false;
+        let changed = RefCell::new(changed);
```

Comment:
For `Copy` types, we prefer using `std::cell::Cell` over `std::cell::RefCell` (surprised that clippy doesn't warn about this!)